File: 456.cpp
class Solution {
public:
    bool find132pattern(vector<int>& nums) {
        int two=INT_MIN;
        stack<int> s;
        for(int i=nums.size()-1;i>=0;i--){
            if (nums[i]<two){
                return true;
            }else{
                while(!s.empty()&&nums[s.top()]<nums[i]){
                    two=nums[s.top()];
                    s.pop();
                }
                s.push(i);
            }
        }
        return false;
    }
};

File: 456.go
func find132pattern(nums []int) bool {
    for i:=1;i<len(nums)-1;i++{
        min:=nums[i]
        for n:=0;n<i;n++{
            if nums[n]<min{
                min=nums[n]
            }
        }
        if min==nums[i]{
            continue
        }
        for n:=i+1;n<len(nums);n++{
            if nums[n]<nums[i]&&nums[n]>min{
                return true
            }
        }
    }
    return false
}